### PR TITLE
feat: add alter table column set data mask policy support

### PR DESCRIPTION
**Pull request by lichuang**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

feat: add alter table column set data mask policy support

Closes https://github.com/datafuselabs/databend/issues/11598

### CODE REVIEW

File: src/meta/proto-conv/tests/it/v040_table_meta.rs [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+                }),
+                ce::TableField::new(
+                    "array",
+                    ce::TableDataType::Array(Box::new(ce::TableDataType::Boolean)),
+                ),
+                ce::TableField::new("variant", ce::TableDataType::Variant),
+                ce::TableField::new("variant_array", ce::TableDataType::Variant),
+                ce::TableField::new("variant_object", ce::TableDataType::Variant),
+                // NOTE: It is safe to convert Interval to NULL, because `Interval` is never really used.
+                ce::TableField::new("interval", ce::TableDataType::Null),
```
Comment (drmingdrmer) [inline src/meta/proto-conv/tests/it/v040_table_meta.rs:130]: We do not have to test all of the field types in this PR

**Review by drmingdrmer — APPROVED** [review]

File: src/query/ast/src/parser/statement.rs [hunk 2]
```diff
@@ -1733,6 +1733,15 @@ pub fn alter_table_action(i: Input) -> IResult<AlterTableAction> {
         },
         |(_, _, column)| AlterTableAction::AddColumn { column },
     );
+    let modify_column = map(
```
Comment (andylokandy) [inline src/query/ast/src/parser/statement.rs:1736]: Would you like to add a unit test in `src/query/ast/tests/it/parser.rs`?

File: src/query/sql/src/evaluator/block_operator.rs [hunk 3]
```diff
@@ -55,6 +55,9 @@ pub enum BlockOperator {
 
     /// Expand the input [`DataBlock`] with set-returning functions.
     FlatMap { srf_exprs: Vec<Expr> },
+
+    /// Replace the input [`DataBlock`].column(i) with set-returning functions if `exprs[i]` is Some.
```
Comment (andylokandy) [inline src/query/sql/src/evaluator/block_operator.rs:59]: I think it's not set-returning functions

File: src/query/storages/fuse/src/operations/read_data.rs [hunk 4]
```diff
@@ -73,6 +77,42 @@ impl FuseTable {
...
+            let mut exprs = Vec::with_capacity(num_input_columns);
+            for i in 0..num_input_columns {
+                let expr = data_mask_policy
+                    .get(&i)
+                    .map(|remote_expr| remote_expr.as_expr(&BUILTIN_FUNCTIONS));
+
+                exprs.push(expr);
+            }
+
+            let op = BlockOperator::Replace { exprs };
```
Comment (andylokandy) [inline src/query/storages/fuse/src/operations/read_data.rs:97]: There is no need for an additional `BlockOperator::Replace`; a `BlockOperator::Map` will perfectly do the job

Comment (lichuang) [reply src/query/storages/fuse/src/operations/read_data.rs]: ok, use `Map` and `Project` instead

File: src/query/storages/fuse/src/operations/read_data.rs [hunk 5]
```diff
@@ -83,18 +85,39 @@ impl FuseTable {
         plan: &DataSourcePlan,
         pipeline: &mut Pipeline,
     ) -> Result<()> {
+        // internal function `walk_expr` is used to modify arg column id to schema column id,
+        // Otherwise `Evaluator::check_expr` may assert fail.
+        fn walk_expr<Index: ColumnIndex>(expr: &mut Expr<Index>, index: Index) {
```
Comment (andylokandy) [inline src/query/storages/fuse/src/operations/read_data.rs:90]: Consider using `Expr::project_column_ref()`

**Review by andylokandy — APPROVED** [review]

**Review by drmingdrmer — APPROVED** [review]

**Review by drmingdrmer — APPROVED** [review]

**Pull request merged**